metabase/metabase
# Check CSV Upload permissions on the frontend

Closes https://github.com/metabase/metabase/issues/30111

### Description

This should properly show CSV upload UI based on a user's permissions and instance settings.

Namely: a user can upload a CSV if:
1. Uploads are enabled and configured on the instance, and
2. The user has "curate" permissions on the collection they're uploading to, AND
3. The user has data access to the database they're uploading to

User Cannot Upload | User Can Upload
---|---
![Screen Shot 2023-04-13 at 2 34 55 PM](https://user-images.githubusercontent.com/30528226/231877401-cf1f43f7-b757-466f-b42d-c321537d734c.png) | ![Screen Shot 2023-04-13 at 2 35 02 PM](https://user-images.githubusercontent.com/30528226/231877396-41eead2f-f6da-4947-a003-1b5bcaf12484.png)


### How to verify

- In metabase enterprise edition, set up a user who has view-only access to a collection
- see that the limited user cannot upload to that collection
- Turn on/off uploads in admin settings and see that the upload button and dropzone get enabled/disabled appropriately for admin users


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/30094)
